src/sources.rs

```diff
@@ -61,7 +61,7 @@ impl Sources {
         // Build our Source object.
         let source = Source {
             alias: alias.clone(),
-            context: context.clone(),
+            context: context.without_subdirectory(),
```

Comment:
If your project has two services pointing at different subdirectories of the same repo, that should only result in a single `Source` object. This change makes it so that there will be a single `Source`, with a context of only the common part. Without this change, you would either get two different conflicting sources, or you'd get a context with the subdirectory of whichever service it processed first, which doesn't make any sense.

This change also makes it so that `cage source clone` (and probably other actions) will use a cleaner URL, without the subdirectory part. I have no idea what would happen if you try to clone a git URL with a subdirectory in it.

Reply:
This is a great comment. In fact, let's put it in the source code. :-)